lib/ansible/modules/cloud/vmware/vmware_vcenter_settings.py

```diff
@@ -241,8 +241,9 @@ def ensure(self):
         directory_query_limit_size = self.params['user_directory'].get('query_limit_size')
         directory_validation = self.params['user_directory'].get('validation')
         directory_validation_period = self.params['user_directory'].get('validation_period')
-        mail_server = self.params['mail'].get('server')
-        mail_sender = self.params['mail'].get('sender')
+        mail = self.params.get('mail') or {'mail': {'server': '', 'sender': ''}}
+        mail_server = mail.get('server', '')
```

Comment:
Nitpicking, you don't need the `get()` since you initialize the mail dict above the a 'server' key:

```python
mail_server = mail['server']
mail_sender = mail['send']
```